src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
+/* eslint-disable */
 
 class Animal {
+  static newBeast(contecst) {
+    this.alive.push(contecst)
+  }
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.newBeast(this);
+  }
   // write your code here
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (!beast.hidden && beast instanceof Carnivore === false) {
```

Comment:
Looks more logical this way
```suggestion
    if (beast instanceof Herbivore && !beast.hidden) {
```